wordpress-mobile/WordPress-iOS
# Improvements to My Site tab.

Swaps the list of sites and blog details when the `newNavBarAppearance` flag is on.

Should not change anything when the flag is off.

## Known limitations:

When the `newNavBarAppearance` feature flag is ON there's currently no way to switch sites.  I've disabled it temporarily until I can implement the site switching.

## To test:

The most important testing that can be done is when the feature flag `newNavBarAppearance` is off, since that's the experience that our users will have in the release version of the App.

Please take a moment to thoroughly test and review the code in this PR.

### Test 1:

Feature flag `newNavBarAppearance` ON.
Use an account that has at least 1 blog.

1. Tap "My Site" in the tab bar
2. Make sure the first VC shown the blog details for your last used blog, or your main blog.
3. Make sure there's no option to go to the list of sites just yet (coming up in my next PR).

### Test 2:

Feature flag `newNavBarAppearance` ON.
Use an account that has no blogs.

1. Tap "My Site" in the tab bar
2. Make sure you see a VC that says you have no blogs.
3. Create a site, make sure once you're done you see the blog details for the blog you created.
4. Delete the site, make sure once you're done you see a no-results view stating you have no blogs.

### Test 3:

Feature flag `newNavBarAppearance` OFF.
Use an account that has at least 1 blog.

1. Tap "My Site" in the tab bar
2. Make sure the first VC shown the blog details for your last used blog, or your main blog.
3. Make sure you can tap on the back button in the navigation bar to pick a different site.
4. Switch to another site and make sure it works fine.

### Test 4:

Feature flag `newNavBarAppearance` OFF.
Use an account that has no blogs.

1. Tap "My Site" in the tab bar
2. Make sure the first VC shown is a no-results saying you have no sites.
3. Create a site.
4. Make sure you see the list of blogs with the site you created.
5. Delete the site.
6. Make sure you see the no results view again.

### Additional notes:

In all tests, make sure you also tap in the user's icon, and that the Me VC comes up just fine.

## PR submission checklist:
